sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/Metrics.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Metrics;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Azure.Messaging.EventHubs.Stress
+{
+    internal class Metrics
+    {
+        public TelemetryClient Client;
+
+        public string CorruptedPropertiesEvent = "CorruptedPropertiesEvent";
+        public string CorruptedBodyEvent = "CorruptedBodyEvent";
+        public string LostEventDataEvent = "LostEventDataEvent";
+
+        // Environment statistics
+        public MetricIdentifier MemorySamples = new MetricIdentifier("MemorySamples");
+        public MetricIdentifier GenerationZeroCollections = new MetricIdentifier("GenerationZeroCollections");
+        public MetricIdentifier GenerationOneCollections = new MetricIdentifier("GenerationOneCollections");
+        public MetricIdentifier GenerationTwoCollections = new MetricIdentifier("GenerationTwoCollections");
+
+        // Publisher statistics
+        public MetricIdentifier BatchesCount = new MetricIdentifier("BatchesCount");
```

Comment:
Wouldn't we want to have distinct names for each scenario?   Otherwise, they'd bleed over, no?

Reply:
@jsquire I thought about this too, but the log entries have a column that says which test it comes from, so I went with simplifying the metrics class and filtering it on the other end, but I'm not attached to doing it one way or the other. Is it confusing to have the tests share metrics names?

I think it is easier to combine metrics graphs across tests if they have their own name, if that seems like a useful visualization? I'm not doing that right now. Another option is to include a metrics dimension that includes which test it is, but I'm not sure how much more helpful that is than having it in a separate column.

Reply:
Nice.   I didn't realize that we already had structured data for that.   Good by me - that covers my concern!